StenosisMeasurement3D/qSlicerStenosisMeasurement3DModuleWidget.cxx

```diff
@@ -87,7 +118,20 @@ void qSlicerStenosisMeasurement3DModuleWidget::setup()
                    this, SLOT(onFiducialNodeChanged(vtkMRMLNode*)));
   QObject::connect(d->inputFiducialSelector, SIGNAL(nodeAddedByUser(vtkMRMLNode*)),
                    this, SLOT(onFiducialNodeChanged(vtkMRMLNode*)));
-  
+  QObject::connect(d->inputSegmentSelector, SIGNAL(currentNodeChanged(vtkMRMLNode*)),
+                   this, SLOT(onSegmentationNodeChanged(vtkMRMLNode*)));
+  QObject::connect(d->inputSegmentSelector, SIGNAL(currentSegmentChanged(QString)),
+                   this, SLOT(onSegmentIDChanged(QString)));
+  QObject::connect(d->wallModelSelector, SIGNAL(currentNodeChanged(vtkMRMLNode*)),
+                   this, SLOT(onWallModelNodeChanged(vtkMRMLNode*)));
+  QObject::connect(d->lumenModelSelector, SIGNAL(currentNodeChanged(vtkMRMLNode*)),
+                   this, SLOT(onLumenModelNodeChanged(vtkMRMLNode*)));
+  QObject::connect(d->outputTableSelector, SIGNAL(currentNodeChanged(vtkMRMLNode*)),
+                   this, SLOT(onTableNodeChanged(vtkMRMLNode*)));
+
+  QObject::connect(d->parameterSetSelector, SIGNAL(currentNodeChanged(vtkMRMLNode*)),
+          this, SLOT(setParameterNode(vtkMRMLNode*)));
+
   // Put p1 and p2 ficucial points on the tube spline at nearest point when they are moved.
```

Comment:
We always need to consider the GUI as optional. The module should be fully functional without instantiating a GUI widget. Therefore, you need to put in the logic things like observing some node and updating the scene content when the observed node is modified.

There are convenience features in module logic classes that make it relatively simple to observe modifications of all nodes of a certain type (see `vtkSetAndObserveMRMLNodeEventsMacro` and `ProcessMRMLNodesEvents` and how it is used for example in `vtkSlicerMarkupsLogic`).

Reply:
Using `vtkObserveMRMLNodeEventsMacro` in `ProcessMRMLNodesEvents`

How does it differ from `vtkSetAndObserveMRMLNodeEventsMacro`?

Reply:
vtkSetAndObserveMRMLNodeEventsMacro is useful when you want to keep a pointer to a node and observe events that the node invokes. The macro does everything (removes observations from the previously observed node, releases reference from the old node, adds observations and reference to the newly selected node).

vtkObserveMRMLNodeEventsMacro just adds observations to a node.